### PR TITLE
feat(locksmith ) - refactor bulk metadata with merged fields

### PR DESCRIPTION
**Pull request by kalidiagne**
# Description
- refactor bulk metadata with merged fields

# Issues

Fixes #
Refs #

# Checklist:

- [x] 1 PR, 1 purpose: my Pull Request applies to a single purpose
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have updated the docs to reflect my changes if applicable
- [ ] I have added tests (and stories for frontend components) that prove my fix is effective or that my feature works
- [x] I have performed a self-review of my own code
- [ ] If my code involves visual changes, I am adding applicable screenshots to this thread

### CODE REVIEW

File: locksmith/src/controllers/v2/metadataController.ts [hunk 1]
```diff
@@ -390,15 +391,40 @@ export class MetadataController {
...
-      const results = await KeyMetadata.findAll({
-        where: {
-          address: lockAddress,
-          chain: network,
-        },
+      const mergeKeysMetadata = keys.map(async (key: any) => {
+        const mergedData = key?.keys?.map(async ({ owner, keyId }: any) => {
+          const keyOwner: string = owner.address
+          const keyHolderMetadata =
+            await lockOperations.getKeyHolderMetadataByAddress(
```
Comment (julien51) [inline locksmith/src/controllers/v2/metadataController.ts:408]: I am not a huge fan of doing an SQL request for _each_, but I think that's a start.

File: locksmith/src/controllers/v2/metadataController.ts [hunk 2]
```diff
@@ -390,15 +391,53 @@ export class MetadataController {
...
       const lockAddress = Normalizer.ethereumAddress(request.params.lockAddress)
       const network = Number(request.params.network)
+      const [keys]: any = request.body
+
+      if (!keys?.keys) {
+        return response
+          .send({
+            message: 'Parameter `keys` is not present',
+          })
+          .status(500)
```
Comment (julien51) [inline locksmith/src/controllers/v2/metadataController.ts:401]: Let's use a 400 code

**Review by julien51 — COMMENTED** [review]
Ok looks good.
Have you tested this thoroughly?

**Comment by kalidiagne** [comment]
> Ok looks good. Have you tested this thoroughly?

Added one more test and tested different scenarios!

**Review by julien51 — APPROVED** [review]

**Pull request merged**